src/occupation.jl

```diff
@@ -71,6 +71,16 @@ function compute_occupation(basis::PlaneWaveBasis{T}, eigenvalues;
                              Roots.Bisection(), atol=eps(T))
     end
 
+    if iszero(temperature)
+        # At zero temperature, we make sure that the Fermi level lies strictly
+        # inside the band gap
+        HOMO = maximum(maximum.(filter.(x->x<εF, eigenvalues)))
+        HOMO = mpi_max(HOMO, basis.comm_kpts)
+        LUMO = minimum(minimum.(filter.(x->x≥εF, eigenvalues)))
+        LUMO = mpi_min(LUMO, basis.comm_kpts)
+        εF = ( HOMO + LUMO ) / 2
+    end
+
     if !isapprox(compute_n_elec(εF), n_electrons)
```

Comment:
true!

Reply:
That's not just the case here. The branching above with the root finding also does not make sense. Just think through the logic of the whole function and simplify.

Reply:
I'm not sure I understand what you mean. Rewriting the whole function with a branching `iszero(temperature)` at the beginning to avoid the bisection in this case ?

Reply:
Ok, thanks for the clarification, I can indeed do that.

Reply:
> Rewriting the whole function with a branching

Not necessarily just one branch, but just try to unify branches and get rid of some that are not actually helpful, since their work is superseeded later. But keep an eye on edge cases, there are quite a few that can occur.